src/librustdoc/core.rs

```diff
@@ -187,16 +187,33 @@ pub fn run_core(search_paths: SearchPaths,
         _ => None
     };
 
-    let warning_lint = lint::builtin::WARNINGS.name_lower();
+    let intra_link_resolution_failure_name = lint::builtin::INTRA_DOC_LINK_RESOLUTION_FAILURE.name;
+    let warnings_lint_name = lint::builtin::WARNINGS.name;
+    let lints = lint::builtin::HardwiredLints.get_lints()
+                    .iter()
+                    .chain(rustc_lint::SoftLints.get_lints())
+                    .filter_map(|lint| {
+                        if lint.name == warnings_lint_name ||
+                           lint.name == intra_link_resolution_failure_name {
+                            None
```

Comment:
Do we really want to be excluding `warnings` in the lints being `allow`ed here? It seems to be doing the opposite of the old code.

Reply:
It's the opposite, we're allowing everything.

Reply:
Are we sure that allowing a bunch of lints individually will act the same as allowing `warnings` all at once? Can we be sure that all warnings that can occur during a rustdoc run will always be placed in one of the lists you're chaining here?

Reply:
If we set all warnings at once using this one, then the intra-doc lint stop working.

Reply:
Ohhhhh, that makes sense, i forgot about that. In that case, this is fine.